app/views.py

```diff
@@ -327,16 +397,34 @@ def forum_post(groupID, threadID):
     posts = Post.query.filter_by(threadID=threadID).all()
     users = get_post_users(posts)
 
-    form = PostForm()
+    postForm = PostForm()
+    delThreadForm = DeleteForm(prefix="thread")
+    delPostForm = DeleteForm(prefix="post")
 
     # POST request for new post
-    if form.validate_on_submit():
-        save_post(form, threadID)
+    if postForm.validate_on_submit():
+        save_post(postForm, threadID)
         flash('Your post is now live!')
         return redirect(url_for('forum_post',groupID=groupID,threadID=threadID))
     
     # GET request for forum thread
-    return render_template('posts.html', title=' | Forum', thread=thread,posts=posts, form=form, users=users)
+    return render_template('posts.html', title=' | Forum', thread=thread,posts=posts, postForm=postForm, delThreadForm=delThreadForm, delPostForm=delPostForm, users=users)
+
+@app.route('/class/<int:groupID>/forum/thread/<int:threadID>/delete', methods=['POST'])
+def delete_thread(groupID, threadID):
+    # Check validity of link access first
+    group = validate_group_link(groupID)
+    thread = Thread.query.filter_by(groupID=groupID,id=threadID).first_or_404()
+
+    postForm = PostForm()
+    delThreadForm = DeleteForm(prefix="thread")
+    delPostForm = DeleteForm(prefix="post")
+
+    if delThreadForm.validate_on_submit():
+
+        remove_thread(thread)
+        flash('Thread deleted')
+        return redirect(url_for('forum', groupID=groupID))
```

Comment:
Can students delete other forum posts as well, out of the ones they created?

Reply:
In `posts.html`, only the educator can delete threads for now. Students can delete their own posts only, under `delete_post()`.